Document and initiate new logging strategy to return log file content. (#655)
kubearchive/kubearchive: docs/modules/ROOT/pages/design/logging-integration.adoc

`Pod` being archived. Earlier archival requests may have already created records
in the `log_url` table, and they should be removed to avoid duplicates.

> The sink will also need to mount and use the `kubearchive-logging` `ConfigMap`

Reviewer: ```suggestion
The sink will mount and use the `kubearchive-logging` `ConfigMap`
```

Author: Fixed.

----
ka logs resource name
----
This command will return the log contents for the the default container in the `Pod`.
example:
[source,bash]
----
ka logs PipelineRun generate-logs-9fkp8 -n generate-logs-pipelines -c generate
----
This will return the log URLs associated with the `PipelineRun` named "generate-logs-9fkp8"
for the container "generate".

The CLI will have to traverse the owner references to gather all of the logs associated
with the given resource. Note that logs could be queried for any resource. Most
resource will not have any logs associated with them or their descendents, but queries on
logs for `TaskRuns` and `Pods` are possible.

> The CLI will also have to perform post-processing on the output from the response to

Reviewer: ```suggestion
The CLI will perform post-processing on the output from the response to
```

Author: Fixed.

log URL in order to get the actual log output. This post-processing is a JSONPath expression
to be applied to the response body.  This post-processing only needs to be done if the
pre-defined entry `LOG_URL_JSONPATH` is contained in the logging `ConfigMap`.

=== API

> The API needs to providate a mechanism similar to the CLI for retrieving log file contents.

Reviewer: ```suggestion
The API will provide a mechanism similar to the CLI for retrieving log file contents.
```

Author: Fixed.
